lib/diff.ml

```diff
@@ -128,8 +131,13 @@ let module_type_fallback ~loc ~typing_env ~name ~reference ~current =
       Some (Module { mname = name; mdiff = Modified Unsupported })
 
 let type_expr typing_env reference current =
-  if Ctype.does_match typing_env reference current then None
-  else Some (Modified { reference; current })
+  let equal =
+    match (get_desc reference, get_desc current) with
+    | Tvar (Some ref_name), Tvar (Some cur_name) ->
+        Ctype.does_match typing_env reference current && ref_name = cur_name
+    | _, _ -> Ctype.does_match typing_env reference current
+  in
```

Comment:
Why do you make a special case here?
For instance, if `reference` is `'a` and `current` is `'b` we get the special case, but if `reference` is `'a -> 'a` and `current` is `'b -> 'b` we do not!

This might be a problem...

Reply:
This is the "hack" I depended on instead of checking the binding of a type variable.. which I wasn't sure how to do that, but @NathanReb  mentioned I can use `Ctype.equal` for comparing type parameters, but I think this is a simpler way...

This special case needs to be expanded to handle other type descs (including Tarrow). I didn't do the other cases since I wasn't sure if this were the best way to handle the binding of type variables..
I will add a test to exploit that, and handle the other cases.